lib/pages/chat/events/sending_video_widget.dart

```diff
@@ -0,0 +1,166 @@
+      sendingFileProgressNotifier.value = 1;
+    }
+
+    if (isPlay) {
+      sendingFileProgressNotifier.value = -1;
+    }
+
+    return ValueListenableBuilder<double>(
+      key: ValueKey(widget.event.eventId),
+      valueListenable: sendingFileProgressNotifier, 
+      builder: ((context, value, child) {
+        if (value == -1) {
+          return child!;
+        }
+        return Stack(
+          alignment: Alignment.center,
+          children: [
+            child!,
+            if (value == 0)... [
+              Container(
+                width: 56,
+                height: 56,
+                decoration: const BoxDecoration(
+                  color: Colors.black38,
+                  shape: BoxShape.circle,
+                ),
+                child: CircularProgressIndicator(
+                  strokeWidth: 2,
+                  color: LinagoraRefColors.material().primary[100],
+                ),
+              ),
+              Icon(
+                Icons.play_arrow, 
+                color: LinagoraRefColors.material().primary[100],
+                size: 48,
+              ), 
+            ] else if (value == 1)... [
+              InkWell(
+                onTap: _onPlayVideo,
+                child: Container(
+                  width: 56,
+                  height: 56,
+                  decoration: const BoxDecoration(
+                    color: Colors.black38,
+                    shape: BoxShape.circle,
+                  ),
+                  alignment: Alignment.center,
+                  child: Icon(
+                    Icons.play_arrow_rounded,
+                    color: LinagoraRefColors.material().primary[100],
+                    size: 48,
+                  ),
+                ),
+              )
+            ]
+          ],
+        );
+      }),
+      child: InkWell(
+        onTap: _onTap,
+        child: ClipRRect(
+          borderRadius: BorderRadius.circular(12.0),
+          child: chewieController != null 
+            ? SizedBox(
+                width: MessageContentStyle.imageBubbleWidth,
+                height: MessageContentStyle.imageBubbleHeight,
+                child: FittedBox(
+                  fit: BoxFit.fitWidth,
+                  child: IgnorePointer(child: Chewie(controller: chewieController!)),
+                ),
+              )
+            : Image.memory(
+              widget.bytes,
+              width: MessageContentStyle.imageBubbleWidth,
+              height: MessageContentStyle.imageBubbleHeight,
+              fit: BoxFit.cover,
+              filterQuality: FilterQuality.medium,
+            ),
+        ),
+      ),
+    );
+  }
+
+  void _onTap() {
+    chewieController?.toggleFullScreen();
+  }
+
+  void _onPlayVideo() async {
+    if (chewieController == null || videoPlayerController == null) {
+      await initVideoController();
+    }
+    setState(() {
+      chewieController?.play();
+      isPlay = true;
+    });
```

Comment:
IMO: Don't need setState here

```suggestion
    setState(() {
      chewieController?.play();
      sendingFileProgressNotifier.value = -1;
    });
```